client/galaxy/scripts/components/Workflow/Run/WorkflowRun.vue

```diff
@@ -0,0 +1,162 @@
+        preferSimpleForm: {
+            type: Boolean,
+            default: false
+        }
+    },
+    data() {
+        return {
+            error: null,
+            loading: true,
+            hasUpgradeMessages: false,
+            hasStepVersionChanges: false,
+            workflowName: "",
+            runForm: null,
+            runButtonEnabled: true,
+            runButtonWaitText: "",
+            runButtonPercentage: -1,
+            invocations: null,
+            simpleForm: null,
+            model: null
+        };
+    },
+    created() {
+        getRunData(this.workflowId)
+            .then(runData => {
+                const model = new WorkflowRunModel(runData);
+                let simpleForm = this.preferSimpleForm;
+                if (simpleForm) {
+                    // These only work with PJA - the API doesn't evaluate them at
+                    // all outside that context currently. The main workflow form renders
+                    // these dynamically and takes care of all the validation and setup details
+                    // on the frontend. If these are implemented on the backend at some
+                    // point this restriction can be lifted.
+                    if (model.hasReplacementParametersInToolForm) {
+                        console.log("cannot render simple workflow form - has ${} values in tool steps");
+                        simpleForm = false;
+                    }
+                    // If there are required parameters in a tool form (a disconnected runtime
+                    // input), we have to render the tool form steps and cannot use the
+                    // simplified tool form.
+                    if (model.hasOpenToolSteps) {
+                        console.log(
+                            "cannot render simple workflow form - one or more tools have disconnected runtime inputs"
+                        );
+                    }
+                    // Just render the whole form for resource request parameters (kind of
+                    // niche - I'm not sure anyone is using these currently anyway).
+                    if (model.hasWorkflowResourceParameters) {
+                        console.log(`Cannot render simple workflow form - workflow resource parameters are configured`);
+                        simpleForm = false;
+                    }
+                }
+                this.simpleForm = simpleForm;
+                this.model = model;
+                this.hasUpgradeMessages = model.hasUpgradeMessages;
+                this.hasStepVersionChanges = model.hasStepVersionChanges;
+                this.workflowName = this.model.name;
+                this.loading = false;
+            })
+            .catch(response => {
+                console.log(response);
+                this.error = errorMessageAsString(response);
+            });
+    },
+    methods: {
+        execute() {
+            this.$refs["run-form"].execute();
```

Comment:
It would be nice if we could develop a cleaner way for the interaction between the vue-component and backbone sub views. Maybe a reusable shim in order to avoid the over use of `$refs`.

Reply:
Thanks for the review @guerler!

From https://vuejs.org/v2/guide/components-edge-cases.html

> $refs are only populated after the component has been rendered, and they are not reactive. It is only meant as an escape hatch for direct child manipulation - you should avoid accessing $refs from within templates or computed properties.
 
This usage is neither within a template or a computed property. I'm not sure I see value in avoiding ``$refs`` here - this seems like a totally fine instance of this. Is there a reason you disagree? I'm trying to understand the concern.

Since ``run-form`` can be one of two forms - I think any property handling thing would lead to duplication in the template stuff but I did look around for ideas nonetheless...

I found this:

https://www.reddit.com/r/vuejs/comments/79stdr/how_to_avoid_refs_here/

The idea I guess there would be that we addd a property "submittingForm" to the parent I suppose (WorkflowRun in this case).  And pass submittingForm to both the child elements. I think that would be kind of yucky - especially because I think we would need to add a ``watch`` to both of the child components?

```
watch: 
  formSubmitting: function(value) {
     if(value) {
        execute();
    }
  }
```

Which seems objectively more code and duplication and subjectively less intuitive to me. The top comment in that reddit thread (only 3 points I'll admit):

> As you demonstrate, the code becomes less clear, and harder to reason about in this instance.
> This seems like a good candidate for calling a specific method on your child component this.$refs.canvas.snapshot(), and having that child component fire an event which the parent captures to get the canvas data.
> The only "reaching in" you're doing is calling an available method on your child.

Maybe a wrapper like you suggest would somehow get around this sort of imitating function calls with properties and events... but I don't see it?

Reply:
Thanks for the detailed explanation. It makes sense to me and is a reusable pattern. A tiny improvement would be to use a name without dash to avoid quotation marks.

Reply:
Fair - happy to make that change.

Reply:
Done - and rebased into #9147.